common/oauth2-proxy/components/istio-m2m/requestauthentication.yaml

```diff
@@ -4,11 +4,39 @@ metadata:
   name: m2m-token-issuer
   namespace: istio-system
 spec:
+  # we only apply to the ingress-gateway because:
+  #  - there is no need to verify the same tokens at each sidecar
+  #  - having no selector will apply to the RequestAuthentication to ALL
+  #    Pods in the mesh, even ones which are not part of Kubeflow
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+
   jwtRules:
-  - forwardOriginalToken: true
-    issuer: M2M_TOKEN_ISSUER_PLACEHOLDER
+  - # The `issuer` must be replaced with a Kustomize patch.
+    issuer: PATCH_ME
+
+    # `forwardOriginalToken` is not strictly required to be true.
+    # there are pros and cons to each value:
+    #  - true: the original token is forwarded to the destination service
+    #          which raises the risk of the token leaking
+    #  - false: the original token is stripped from the request
+    #           which will prevent the destination service from
+    #           verifying the token (possibly with its own RequestAuthentication)
+    forwardOriginalToken: true
+
+    # This will unpack the JWTs issued by Kubernetes into the expected headers.
     outputClaimToHeaders:
-    - claim: sub
-      header: x-auth-request-user
```

Comment:
Why are you dropping x-auth-request-user? Should we not migrate to it in the future?

Reply:
Personally, I don't think we need to migrate to those headers, it's a needless breaking change.

Also, since we don't currently use those headers, including them here is only going to confuse people.